src/Forms/FormField.php

```diff
@@ -1222,32 +1234,30 @@ public function Type()
     }
 
     /**
-     * Utility method to call an extension hook which allows the result of validate() calls to be adjusted
-     *
-     * @param bool $result
-     * @param Validator $validator
-     * @return bool
-     * @deprecated 5.4.0 Use extend() directly instead
+     * Determines whether the field is valid or not based on its value
      */
-    protected function extendValidationResult(bool $result, Validator $validator): bool
+    public function validate(): ValidationResult
     {
-        Deprecation::notice('5.4.0', 'Use extend() directly instead');
-        $this->extend('updateValidationResult', $result, $validator);
+        if (count(func_get_args()) !== 0) {
```

Comment:
```suggestion
        if (func_num_args() !== 0) {
```
If you do insist on keeping this, the above change is cleaner.

Reply:
Updated